### PR TITLE
Add memif mechanism networkservice chain elements

### PR DESCRIPTION
**Pull request by edwarnicke**

### CODE REVIEW

File: pkg/networkservicemesh/mechanisms/memif/client.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+package memif
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/ligato/vpp-agent/api/models/vpp"
+	vpp_interfaces "github.com/ligato/vpp-agent/api/models/vpp/interfaces"
```
Comment (fkautz) [inline pkg/networkservicemesh/mechanisms/memif/client.go]: minor nit from the go style guide:

"Good package names are short and clear. They are lower case, with no under_scores or mixedCaps"

File: pkg/networkservicemesh/mechanisms/memif/server.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+package memif
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/ligato/vpp-agent/api/models/vpp"
+	vpp_interfaces "github.com/ligato/vpp-agent/api/models/vpp/interfaces"
```
Comment (fkautz) [inline pkg/networkservicemesh/mechanisms/memif/server.go]: minor nit, remove _

File: pkg/networkservicemesh/mechanisms/memif/client.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+type memifClient struct {
+	baseDir string
+}
+
+// NewClient provides a NetworkServiceClient chain elements that support the memif Mechanism
+func NewClient(baseDir string) networkservice.NetworkServiceClient {
+	return &memifClient{baseDir: baseDir}
+}
+
+func (m *memifClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*connection.Connection, error) {
```
Comment (fkautz) [inline pkg/networkservicemesh/mechanisms/memif/client.go]: opts isn't used here, is this intentional?

File: pkg/networkservicemesh/mechanisms/memif/client.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	request.MechanismPreferences = append(request.MechanismPreferences, mechanism)
+	conn, err := next.Client(ctx).Request(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	m.appendInterfaceConfig(ctx, conn)
+	return conn, nil
+}
+
+func (m *memifClient) Close(ctx context.Context, conn *connection.Connection, opts ...grpc.CallOption) (*empty.Empty, error) {
```
Comment (fkautz) [inline pkg/networkservicemesh/mechanisms/memif/client.go]: opts isn't used here, is this intentional?

Comment (edwarnicke) [reply pkg/networkservicemesh/mechanisms/memif/client.go]: Fixed

File: pkg/networkservicemesh/mechanisms/memif/client.go (hunk 3 above)
Comment (edwarnicke) [reply pkg/networkservicemesh/mechanisms/memif/client.go]: Fixed

File: pkg/networkservicemesh/mechanisms/memif/server.go (hunk 2 above)
Comment (edwarnicke) [reply pkg/networkservicemesh/mechanisms/memif/server.go]: Fixed

File: pkg/networkservicemesh/mechanisms/memif/client.go (hunk 1 above)
Comment (edwarnicke) [reply pkg/networkservicemesh/mechanisms/memif/client.go]: Fixed

**Pull request merged**